pybossa/view/projects.py

```diff
@@ -3234,14 +3236,30 @@ def ext_config(short_name):
                 form = form_class()
                 if not form.validate():
                     flash(gettext('Please correct the errors', 'error'))
+
                 ext_conf[form_name] = {k: v for k, v in six.iteritems(form.data) if v}
                 ext_conf[form_name].pop('csrf_token', None)     #Fflask-wtf v0.14.2 issue 102
-                project.info['ext_config'] = ext_conf
+                
+                if u'gigwork_poller' in ext_conf:
+                    if u'target_bucket' not in ext_conf[u'gigwork_poller']:
+                        project.info[u'ext_config'].pop(u'gigwork_poller')
+                    else:
+                        project.info[u'ext_config'].update({u'gigwork_poller': ext_conf[u'gigwork_poller']})
+
+                if u'hdfs' in ext_conf:
+                    if not u'path' in ext_conf[u'hdfs']:
+                        project.info[u'ext_config'].pop(u'hdfs')
+                    else:
+                        project.info[u'ext_config'].update({u'hdfs': ext_conf[u'hdfs']})   
+
+                if project.info[u'ext_config'] == {}:
+                    project.info.pop(u'ext_config')
+
```

Comment:
Seems what you're trying to achieve here is removing ext_config when no poller or hdfs path specified. This can be simplified to something similar to,
```python
target_bucket = ext_config.get('gigwork_poller', {}).get('target_bucket')
if not target_bucket:
    ext_config.pop('gigwork_poller', None)
hdfs_path = ext_config.get('hdfs', {}).get('path')
    ext_config.pop('hdfs', None)
if ext_config:
     project.info['ext_config'] = ext_conf
```

Also, in case of private instance, it is required to have one of the value. Seems flash error message in such case and failing to update would be helpful.

Reply:
Should this requirement be checked in the form validation instead?
```

if not form.validate():
                    flash(gettext('Please correct the errors', 'error'))
```

Reply:
I'm fine with wherever you think check for private instance and then validation can be applied appropriately.